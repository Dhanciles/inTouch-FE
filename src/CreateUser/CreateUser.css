form {
  display: flex;
  flex-direction: column; 
  justify-content: space-around;
  border: none;  
  padding: 8px; 
  background-color: #F2F3F9;
}

input {
  height: 70px; 
  width: 200px; 
  border-bottom: 2px solid #191C69;
  line-height: 1rem; 
  font-family: 'Oswald', sans-serif;
  font-weight: 300; 
  font-size: 1.5rem;
  color: #191C69;  
}

::placeholder {
  color: #191C69; 
  font-size: 1.5rem; 
  margin-bottom: 6px; 
}

.new-account-button {
  height: 40px; 
  width: 150px; 
  background-color: #191C69; 
  padding: 5px; 
  margin-top: 15px; 
  border-radius: 25px; 
  cursor: pointer;
  color: #F2F3F9;
  font-family:'Open Sans', sans-serif;
  font-weight: 400; 
  font-size: 1rem; 
}

.new-account-button:hover {
  background-color: #ABAECF; 
  color: #5B609F; 
  cursor: pointer;
}
