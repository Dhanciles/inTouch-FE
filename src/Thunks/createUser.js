import { isLoading, hasErrored, signInUser } from '../actions/index.js'
import { authenticateUser }  from './authenticateUser'
require('isomorphic-fetch');
export const createUser = (userName, email, password) => {
  debugger;
  const url = 'https://in-touch-dev.herokuapp.com/api/v1/data/'
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url, {
        method: 'POST',
        body: JSON.stringify({query: `mutation { createUser(email:"${email}", username:"${userName}", password:"${password}"){user {username}}}`}),
        headers: {
          'Content-Type': 'application/json'
        }
      })
      if (!response.ok) {
        dispatch(isLoading(false))
        throw Error(response.statusText)
      }
        dispatch(isLoading(false))
      const data = await response.json()
      const name = data.data.createUser.user.username
      debugger;
      const token = await dispatch(authenticateUser(name, password))
      debugger; 
        dispatch(signInUser(name, token))
    } catch (error) {
        dispatch(hasErrored(true))
    }
  }
}