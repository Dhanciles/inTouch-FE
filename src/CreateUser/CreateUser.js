import React, {Component} from 'react'; 
import './CreateUser.css'

export default class CreateUser extends Component {
  constructor() {
    super()
    this.state = {
      name: '', 
      email: '', 
      password: '',
      confirmation: ''
    }
  }

  render() {
    const { name, email, password, confirmation } = this.state
    return (
      <form>
        <fieldset>
          <input className="create-user-input" name="name" value={name} type="text" required placeholder="Name"/>
          <input className="create-user-input" name="email" value={email} type="text" required placeholder="Email"/>
          <input className="create-user-input" name="password" value={password} type="password" required placeholder="Password"/>
          <input className="create-user-input" name="confirmation" value={confirmation} type="password" required placeholder="Confirmation"/>
        </fieldset>
        <button className="create-account-button">Create Account</button>
      </form>
    )
  }
}