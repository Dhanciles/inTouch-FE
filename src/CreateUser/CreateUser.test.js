import React from 'react'; 
import { shallow } from 'enzyme'; 
import { CreateUser, mapDispatchToProps } from './CreateUser.js'; 
import { createUser } from '../Thunks/createUser.js'


describe('CreateUser', () => {
  let mockName; 
  let mockEmail; 
  let mockPassword; 
  let mockConfirmation; 
  let mockEvent
  let wrapper
  
  beforeEach(() => {
    wrapper = shallow(<CreateUser createNewUser={jest.fn()}/>)
    mockName = 'Kaytranada'; 
    mockEmail = 'yaboy@ymail.com'; 
    mockPassword = 'password'; 
    mockConfirmation = 'password';
    mockEvent = {
     target: {
       name: 'userName',
       value: 'Derek'
      }
    }
  })
  
  it('should match snapshot', () => {
    
    expect(wrapper).toMatchSnapshot()
  })
  
  it('should invoke handleChange when event occurs on an input', () => {
    wrapper.handleChange = jest.fn()
    wrapper.find('.new-user-name').simulate('change', mockEvent)
    
    expect(wrapper.handleChange).toHaveBeenCalled
  })
    
  it('should update state when handleChange is invoked on an input', () => {
    const expected = {
      userName: 'Derek', 
      email: '', 
      password: '',
      confirmation: '', 
      disabled: true
    }
    
    wrapper.instance().handleChange(mockEvent)
    
    expect(wrapper.state()).toEqual(expected)
  })
  
  it('should invoke enableButton', () => {
    const mockFunction = jest.fn()
    wrapper.enableButton = mockFunction
    
    wrapper.instance().handleChange(mockEvent)
    
    expect(mockFunction).toHaveBeenCalled
  })
})