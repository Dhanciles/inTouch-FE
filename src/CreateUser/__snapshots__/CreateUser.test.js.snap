// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateUser should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateUser />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="new-user-name"
          name="name"
          placeholder="Name"
          required={true}
          type="text"
          value=""
        />,
        <input
          className="new-user-email"
          name="email"
          placeholder="Email"
          required={true}
          type="text"
          value=""
        />,
        <input
          className="new-user-password"
          name="password"
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />,
        <input
          className="new-user-password-confirmation"
          name="confirmation"
          placeholder="Confirmation"
          required={true}
          type="password"
          value=""
        />,
        <button
          className="new-account-button"
        >
          Create Account
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "new-user-name",
          "name": "name",
          "placeholder": "Name",
          "required": true,
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "new-user-email",
          "name": "email",
          "placeholder": "Email",
          "required": true,
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "new-user-password",
          "name": "password",
          "placeholder": "Password",
          "required": true,
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "new-user-password-confirmation",
          "name": "confirmation",
          "placeholder": "Confirmation",
          "required": true,
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Create Account",
          "className": "new-account-button",
        },
        "ref": null,
        "rendered": "Create Account",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="new-user-name"
            name="name"
            placeholder="Name"
            required={true}
            type="text"
            value=""
          />,
          <input
            className="new-user-email"
            name="email"
            placeholder="Email"
            required={true}
            type="text"
            value=""
          />,
          <input
            className="new-user-password"
            name="password"
            placeholder="Password"
            required={true}
            type="password"
            value=""
          />,
          <input
            className="new-user-password-confirmation"
            name="confirmation"
            placeholder="Confirmation"
            required={true}
            type="password"
            value=""
          />,
          <button
            className="new-account-button"
          >
            Create Account
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "new-user-name",
            "name": "name",
            "placeholder": "Name",
            "required": true,
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "new-user-email",
            "name": "email",
            "placeholder": "Email",
            "required": true,
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "new-user-password",
            "name": "password",
            "placeholder": "Password",
            "required": true,
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "new-user-password-confirmation",
            "name": "confirmation",
            "placeholder": "Confirmation",
            "required": true,
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Create Account",
            "className": "new-account-button",
          },
          "ref": null,
          "rendered": "Create Account",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
